trade-finance-manager-ui/server/controllers/utilisation-reports/add-to-an-existing-payment/index.ts

```diff
@@ -25,7 +27,12 @@ export const addToAnExistingPayment = async (req: AddToAnExistingPaymentRequest,
     const checkedCheckboxIds = getPremiumPaymentsCheckboxIdsFromObjectKeys(req.body);
     const feeRecordIds = getFeeRecordIdsFromPremiumPaymentsCheckboxIds(checkedCheckboxIds);
 
-    const selectedFeeRecordDetails = await api.getSelectedFeeRecordsDetails(reportId, feeRecordIds, userToken);
+    const selectedFeeRecordDetails = await api.getSelectedFeeRecordsDetailsWithAvailablePaymentGroups(reportId, feeRecordIds, userToken);
+    const { availablePaymentGroups } = selectedFeeRecordDetails;
+
+    if (!availablePaymentGroups || availablePaymentGroups.length === 0) {
+      throw new Error('No available payment groups attached to fee record details response.');
+    }
```

Comment:
I know you are not done writing tests yet, would be good to get some tests for this error case